apps/test-moisture/src/main.cpp

```diff
@@ -7,35 +7,32 @@ DigitalOut led1(LED1);
 int main() {
   led1 = 0;  // start with LED off
 
-  uint16_t moisture = 0;  // initialize default variables
+  float moisture = 0;  // initialize default variables
   float temperature = 0.0;
-  uint8_t HW_ID     = 0;
 
   MoistureSensor sensor = MoistureSensor(TEMP_MOIST_I2C_SDA, TEMP_MOIST_I2C_SCL);
 
   while (1) {
-    sensor.Reset_Sensor();  // reset all registers on sensor to default values
+    sensor.resetSensor();  // reset all registers on sensor to default values
 
     ThisThread::sleep_for(500ms);
 
-    led1 = (sensor.Is_Initialized());  // turn on LED if the sensor's HW_ID code matches the known value
+    led1 = (sensor.getSensorStatus());  // turn on LED if the sensor's HW_ID code matches the known value
 
     printf("\r\nChecking Device ID...\r\n");  // read device HW_ID, reading of 85 is expected
-    HW_ID = sensor.Read_HW_ID();
-    printf("HW ID: %d \r\n", HW_ID);
 
     ThisThread::sleep_for(100ms);
 
     printf("\r\nReading Moisture...\r\n");  // read moisture from sensor, reading of 65534 indicates unsuccessful
                                             // initialization
-    moisture = sensor.Read_Moisture();
-    printf("Moisture: %d \r\n", moisture);
+    moisture = sensor.primaryRead();
+    printf("Moisture: %f \r\n", moisture);
 
     ThisThread::sleep_for(100ms);
 
     printf("\r\nReading Temperature...\r\n");  // read temperature from sensor, reading of -273.0 indicates
                                                // unsuccessful initialization
-    temperature = sensor.Read_Temperature();
+    temperature = sensor.alternateRead();
```

Comment:
Interesting that you changed moisture and temperature reads to primary and alternate, but it could be argued that this is less clear to the user.

Reply:
Correct me if I'm wrong but I believe the idea is to abstract away "sensor" type classes to all have the same interface defined in the base Sensor class. It makes naming schemes more generic but it allows all sensor types to be grouped under one common interface.

Reply:
Yeah things are definitely less clear now but I think we will use object names which are more expressive and can help provide information and context while allowing for a central Sensor Class. @younesr1 I do feel that these changes may be a bit strange and can affect clarity. Do you feel that the trade off is worth it?

Reply:
hey sorry i didnt get to this til now cindy is right. we cant have readmoisture() and readtemperature(). felix, you approach is good, but i would rename primaryRead() to read(). keep alternateRead() as such. Semantically, it makes sense for the read() function of a moisture sensor to read the moisture.